File: ts-sdk/src/sdk/models/shared/groupmini.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum GroupMiniGroupTypeEnum {
    ManagedGroup = "managed_group",
    AllUsersGroup = "all_users_group"
}

export enum GroupMiniTypeEnum {
    Group = "group"
}


// GroupMini
/** 
 * A base representation of a group.
**/
export class GroupMini extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=group_type" })
  groupType?: GroupMiniGroupTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=id" })
  id?: string;

  @SpeakeasyMetadata({ data: "json, name=name" })
  name?: string;

  @SpeakeasyMetadata({ data: "json, name=type" })
  type?: GroupMiniTypeEnum;
}

File: sdks/python/src/boxsdk/models/shared/group_mini.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from boxsdk import utils

class GroupMiniGroupTypeEnum(str, Enum):
    MANAGED_GROUP = "managed_group"
    ALL_USERS_GROUP = "all_users_group"

class GroupMiniTypeEnum(str, Enum):
    GROUP = "group"


@dataclass_json
@dataclasses.dataclass
class GroupMini:
    r"""GroupMini
    A base representation of a group.
    """
    
    group_type: Optional[GroupMiniGroupTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('group_type') }})
    id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    type: Optional[GroupMiniTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
